README.md

```diff
@@ -974,6 +979,91 @@ $ grace interpret - <<< './greet.ffg "John"'
 
 Any subexpression can be imported in this way.
 
+#### Imports using URIs
+
+Imports with URIs work similar to the ones using a simple filepath.
+
+Suppose you do not have the `greet.ffg` stored locally but instead it resides
+on a web server: `http://my-cool-domain.code/grace/greet.ffg`
+You could either download it and reference it by its filepath like demonstrated
+in the example above or let the Grace interpreter do the job:
+
+```bash
+$ grace interpret - <<< 'http://my-cool-domain.code/grace/greet.ffg "John"'
+```
+```dhall
+"Hello, John!"
+```
+
+Note that if a particular URI can be handled by the Grace interpreter depends
+on its (compile-time) configuration: Internally it relies on a set of _resolvers_
+that take care of all the things related to networking like downloading, caching
+, verifying the integrity of retrieved file on so on.
+For instance, the motivating example will unfortunately not work out-of-the box
+since the grace executable has no builtin resolver for HTTP (yet).
+
+Grace comes with three builtin resolvers:
+
+1. One to resolve `env://` URIs,
+2. one to resolve `file://` URIs.
+3. and one that delegates the resolving to an external program.
+
+Lets have a look at the `env://` resolver first:
+```bash
+$ MY_VAR='"Hello !"' grace interpret - <<< 'env:///MY_VAR'
+```
+```dhall
+"Hello !"
+```
+
+The resolver replaces '/' in the URIs path with '_':
```

Comment:
I might have missed this since I've only skimmed so far, but why does the resolver replace `_` with `/` for environment variables?

Reply:
That was one part I was not sure about if it is good to keep it: Since it is a convention to namespace environment variables like `GNUPG_*` or `SSH_*` but in URIs we have '/' as a path separator, I thought it was a nice idea to do such a replace so one could write `env:///SSH/SOCKET` (Well, that is a weird example) instead of `env:///SSH_SOCKET`. Note that the latter one works as well.

I admit that might be confusing and a bit to much; I tend to remove it.

Reply:
I removed the slash replacement. It is the saner way after all.